### PR TITLE
Guessing error reporting

### LINKED ISSUE #123: error reporting for formula guessing
it currently does not report syntax errors; stuck with the previous output.

### PR DESCRIPTION
**Pull request by rlisahuang**
Attempt to fix #123

### CODE REVIEW

File: website/src/components/TutorialComponents/runZ3DuoWeb.ts [hunk 1]
```diff
@@ -1,23 +1,37 @@
...
+    const { Context, Z3: Z3Core } = await z3p;
     let Z3 = Context('main');
 
     let output = '';
     let error = '';
     let outputObj;
 
+    const throwIfError = (ctxPtr) => {
+        const errorCode = Z3Core.get_error_code(ctxPtr);
+        console.log(ctxPtr)
```
Comment (NikolajBjorner) [inline website/src/components/TutorialComponents/runZ3DuoWeb.ts:19]: remove console.log before merging

File: website/src/components/TutorialComponents/runZ3DuoWeb.ts [hunk 2]
```diff
@@ -1,23 +1,37 @@
...
+        }
+    }
+
     try {
         const s1 = new Z3.Solver();
         const s2 = new Z3.Solver();
         s1.fromString(user_input);
         s2.fromString(secret_input);
 
+        throwIfError(s1.ptr);
```
Comment (NikolajBjorner) [inline website/src/components/TutorialComponents/runZ3DuoWeb.ts:32]: ptr is the pointer to the context or solver?

Comment (NikolajBjorner) [reply website/src/components/TutorialComponents/runZ3DuoWeb.ts:32]: I think you want s1.ctx.ptr

Comment (NikolajBjorner) [reply website/src/components/TutorialComponents/runZ3DuoWeb.ts:32]: export interface Solver<Name extends string = 'main'> {
  /** @hidden */
  readonly __typename: 'Solver';

  readonly ctx: Context<Name>;
  readonly ptr: Z3_solver;

Comment (NikolajBjorner) [reply website/src/components/TutorialComponents/runZ3DuoWeb.ts:32]: export interface Context<Name extends string = 'main'> {
  /** @hidden */
  readonly ptr: Z3_context;
  /**

Comment (NikolajBjorner) [reply website/src/components/TutorialComponents/runZ3DuoWeb.ts:32]: the error check code takes a context pointer (Z3_context)

**Comment by rlisahuang** [comment]
Fixed. Ready to merge.

**Pull request merged**